### ISSUE TITLE
`--remap` usage and documentation clarification

### ISSUE THREAD

**Issue by bubbletroubles** [lycheeverse/lychee#1305]
I'm trying to use lychee with `--remap` for more than one URL. The documentation is really unclear on the expected usage.

Some examples use a list of URLs to remap, but I've found that doesn't work. For example

`.\lychee.exe  --remap ['https://mysite.com https://mynewsite.com' , 'https://mysecondsite.com https://mynewsecondsite.com'] . `

returns the error
```
Error: Remaps must be of the form '<pattern> <uri>' (separated by whitespace)

Caused by:
    0: Error when using regex engine: regex parse error:
           [https://mysite.com
           ^
       error: unclosed character class
    1: regex parse error:
           [https://mysite.com
           ^
       error: unclosed character class
```

What did work was specifying two `--remap` parameters.

`.\lychee.exe  --remap 'https://mysite.com file://mynewsite.com' --remap 'https://mysecondsite.com file://mynewsecondsite.com' .`

Is this the expected usage of --remap, to specify one per URL to be remapped?

**Comment by sanmai-NL** [comment]
You're supplying TOML markup to a command line parameter. It works the same as other configuration options: either use the CLI syntax (`'<pattern> <uri>'` doesn't include a bracket, but you include one) or use the configuration file with TOML syntax.

**Comment by sanmai-NL** [comment]
@bubbletroubles Can this be closed?

**Comment by bubbletroubles** [comment]
@sanmai-NL  what is the correct syntax for the CLI?

Is it `.\lychee.exe  --remap 'https://mysite.com file://mynewsite.com' --remap 'https://mysecondsite.com file://mynewsecondsite.com' .` (e.g. one --remap flag per site to be remapped? Or is there syntax to use many sites with one --remap flag)

**Comment by sanmai-NL** [comment]
Two `remap` command-line arguments works for you and the source code supports that finding. If that isn't documented, it should be, of course. You can create a batch script or shortcut if you don't want to type so much.

Copy-pasting technical files from websites on Windows causes problems with text encodings, line endings, etc. Moreover, it's prone to omission of some crucial character due to incomplete selection. Also, there could be a byte-order marker (BOM). Editing with Visual Studio Code instead of e.g.,  Windows Notepad allows you to beter check for and control these issues.

Please clone to Git repository and copy the configuration file as a file instead of text contents.

**Comment by mre** [comment]
@bubbletroubles, the first argument is a regular expression pattern, not just a string.
That's what `Remaps must be of the form '<pattern> <uri>' (separated by whitespace)` means.

So to answer your question, if you can find a regular expression pattern that matches all sites that you want to remap, you can do it in one command.
In your case, it looks like you want to replace the protocol from `https` to `file`:

```
.\lychee.exe  --remap 'https://(.*)$ file://$1' .
```

> [!NOTE]  
> I have not tested this myself, so you might need to tweak it a bit, but you get the idea.

You can test your regular expressions with tools like https://rustexp.lpil.uk/:

<img width="1290" alt="image" src="https://github.com/lycheeverse/lychee/assets/175809/8b711026-112b-4688-aace-9352b4658ff2">

Apologies if the documentation is not entirely clear on this. The documentation is also [open source](https://github.com/lycheeverse/lycheeverse.github.io), and we accept pull requests to improve it. If you find that certain use-cases were not covered, I'd be thankful for a contribution over there.

**Comment by bubbletroubles** [comment]
Thanks all for the great insights - I just went to review the documentation again and I found the answer was already documented :) Amazing what a fresh set of eyes looking at something can find.

https://lychee.cli.rs/#/usage/cli?id=repeating-options
> [Repeating Options](https://lychee.cli.rs/#/usage/cli?id=repeating-options)
Options marked with ... can be specified multiple times. For example, --exclude can be specified multiple times to exclude multiple URLs. This is also true for --headers, --include, **--remap**, --scheme, and --exclude-path.
